src/garage/envs/base.py

```diff
@@ -165,6 +165,16 @@ def __getstate__(self):
             return state
         return self.__dict__
 
+    def __setstate__(self, state):
+        """See `Object.__setstate__.
+
+        Args:
+            state (dict): Unpickled state of this object.
+
+        """
+        obj = type(self)(state['_env'], state['_env_name'])
+        self.__dict__.update(obj.__dict__)
```

Comment:
can't you just do `self.__init__(state['_env'], state['_env_name'])`? I think that is easier to read.

Reply:
Yeah, if `__init__` works, it's better to just call that, in case the environment constructs a circular reference (e.g. `gym.CarRacing`).

Reply:
Fixed